### DISCUSSION TITLE
Drop into editor API finalised

### DISCUSSION THREAD

**Discussion by isidorn** [microsoft/vscode-discussions#96] [Announcements]
The text editor drop API lets extensions handle dropping of files or other content into text editors. These drop events can come from within VS Code, such as dragging a file from VS Code's Explorer into a text editor, or can be generated by dropping a file from your operating system into VS Code.

![api-drop](https://user-images.githubusercontent.com/1926584/183876360-f674546f-00de-4527-926d-19de2a764580.gif)

To drop a file into an editor, users can hold down `Shift` while dragging and then drop the file to a specific location in the file. At this point, your extension can choose what to do with the dropped data.

Check out the [drop into editor extension sample](https://github.com/microsoft/vscode-extension-samples/tree/main/drop-on-document) for a simple example of this API in action.

Try out this API and let us know what you think about it. Thanks!

**Comment by FredLL-Avaiga** [comment]
Hi,
Is it possible to drop a TreeViewItem into a CustomTextEditor (ie WebView) ?

**Reply by isidorn** [reply]
No.

**Reply by FredLL-Avaiga** [reply]
Is it in the plan somehow ?

**Comment by phschaad** [comment]
Do I understand correctly that it is not possible to use this with anything other than a plain, standard TextEditor? I.e., you cannot drag into a `CustomTextEditor`, let alone a `CustomEditor`?
